src/app/security-group/components/sg-rule-addition-form/sg-rule-addition-form.component.html

```diff
@@ -0,0 +1,116 @@
+<form id="rule-form" [formGroup]="ruleForm" #mainForm="ngForm" (ngSubmit)="onSubmit()">
+  <mat-form-field>
+    <mat-select class="type" formControlName="type">
+      <mat-option *ngFor="let t of types" [value]="t.value">
+        {{ t.text | translate }}
+      </mat-option>
+    </mat-select>
+  </mat-form-field>
+
+  <mat-form-field>
+    <mat-select formControlName="protocol">
+      <mat-option *ngFor="let p of protocols" [value]="p.value">
+        {{ p.text | translate }}
+      </mat-option>
+    </mat-select>
+  </mat-form-field>
+
+  <mat-form-field class="cidr">
+    <input
+      matInput
+      type="text"
+      formControlName="cidr"
+      [placeholder]="'SECURITY_GROUP_PAGE.RULES.ENTER_CIDR' | translate"
+      required
+    >
+    <mat-error>
+      {{ 'SECURITY_GROUP_PAGE.RULES.ENTER_VALID_CIDR' | translate }}
+    </mat-error>
+  </mat-form-field>
+
+  <!--This container contains start, end port fields for protocols such TCP, UDP-->
+  <ng-container *ngIf="!isIcmpProtocol" formGroupName="params">
+    <mat-form-field>
+      <input
+        matInput
+        type="number"
+        formControlName="startPort"
+        [errorStateMatcher]="portMatcher"
+        [placeholder]="'SECURITY_GROUP_PAGE.RULES.START_PORT' | translate"
+        required
+      >
+      <mat-error>
```

Comment:
Can you please list out all the error messages here and show them using ngIf?
similar to other fields

Reply:
It has already been discussed and decided to leave it as it is